src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -3155,7 +3155,7 @@ private BoundExpression BindArrayDimension(ExpressionSyntax dimension, Diagnosti
                 var size = BindValue(dimension, diagnostics, BindValueKind.RValue);
                 if (!size.HasAnyErrors)
                 {
-                    size = ConvertToArrayIndex(size, dimension, diagnostics, allowIndexAndRange: false);
+                    size = ConvertToArrayIndex(size, dimension, diagnostics, allowIndexAndRange: false, allowNIntAndNUInt: false);
```

Comment:
Are we testing this case? If not, perhaps test:
```C#
class Program
{
    static void Main()
    {
        const nint i = 3;
        const nuint u = 4;
        int[] a;
        a = new int[i];
        a = new int[u];
    }
}
```

Reply:
Added. This looks to be another case where we should update to avoid the:
```
conv.i8
conv.ovf.i
```